lib/tapioca/generator.rb

```diff
@@ -522,5 +547,35 @@ def compile_dsl_rbi(constant, contents)
 
       filename
     end
+
+    sig { params(tmp_dir: Pathname).void }
+    def verify_dsl_rbi(tmp_dir:)
+      existing_rbis = get_file_list(dir: config.outpath).sort
+      test_rbis = get_file_list(dir: tmp_dir).sort
+
+      current_count = existing_rbis.count
+      new_count = test_rbis.count
+
+      raise(OutOfSyncError.new, "New file(s) introduced.") if current_count != new_count
+
+      desynced_files = []
+
+      (0..current_count - 1).each do |i|
+        desynced_files << test_rbis[i] unless FileUtils.identical?(existing_rbis[i], test_rbis[i])
+      end
+
+      if desynced_files.count > 0
+        filenames = desynced_files.map(&:to_s).each do |file|
+          file.sub!(tmp_dir.to_s, "sorbet/rbi/dsl")
```

Comment:
NIT: instead of mapping to strings and then iterating again, can we use a single map and invoke `to_s` on `file` inside the loop?